src/xdg-desktop-portal.c

```diff
@@ -184,6 +185,9 @@ export_portal_implementation (GDBusConnection *connection,
 static void
 peer_died_cb (const char *name)
 {
+#ifdef HAVE_GUDEV
+  revoke_usb_devices_from_sender (name);
```

Comment:
You can call `xdp_connection_track_name_owners` to connect to a callback all in `usb.c` without leaking this detail here.